### PR TITLE
Added the bounded box option

### PR DESCRIPTION
**Pull request by csese**

### CODE REVIEW

**Comment by christophergandrud** [comment]
Thanks. When I run the examples I always get the error:

```{S}
Error in `[.data.frame`(Links, , Source) : undefined columns selected
```

File: inst/htmlwidgets/forceNetwork.js [hunk 1]
```diff
@@ -140,6 +140,22 @@ HTMLWidgets.widget({
...
 
+	// Add the option for a bounded box
+	function nodeBoxX(d,width) {
+		if(options.bounded){
+            var dx = Math.max(nodeSize(d), Math.min(width - nodeSize(d), d.x));		
+			return dx;                    
+        }else{
+            return d.x}                    
+    }
+	function nodeBoxY(d,width) {
```
Comment (ellisp) [inline inst/htmlwidgets/forceNetwork.js]: Should line 151 be height not width?
```javascript
function nodeBoxY(d,height) {
```

**Comment by christophergandrud** [comment]
https://github.com/christophergandrud/networkD3/issues/51

**Comment by KobaKhit** [comment]
There is confusion. A bounded box was a feature request [#51](https://github.com/christophergandrud/networkD3/issues/51) that was implemented here [#52](https://github.com/christophergandrud/networkD3/pull/52). So the box is bounded only if the option `bounded` is set to `TRUE`. My pull request improves the original **optional** bounded box. I am not sure how the zoom will behave when the box is bounded, but again the bounded box is an option that can be switched on or off in the graph function. Let me know if I need to clarify further.

**Pull request merged**